libraries/AP_HAL_ChibiOS/hwdef/scripts/chibios_hwdef.py

```diff
@@ -3107,6 +3117,294 @@ def write_default_parameters(self):
         self.romfs_add('defaults.parm', filepath)
         self.have_defaults_file = True
 
+    def readme_write_intro(self, f):
+        f.write('''
+# %s Flight Controller
+
+The %s is a flight controller designed and produced by [%s].
+''' % (self.board_name, self.board_name, self.board_name))
+        f.write('''
+## Features
+
+ - MCU - %s 32-bit processor running at 480 MHz\n''' % self.mcu_type)
+        count = 1
+        for dev in self.imu_list:
+            f.write(' - IMU%u - %s\n' % (count, dev[0]))
+            count = count+1
+        if self.baro_list:
+            f.write(' - Barometer - %s\n' % self.baro_list[0][0])
```

Comment:
This is a problem if the board has more than one barometer.

And the real problem is that I don't think we can distinguish between boards that have two barometers and boards that have *alternative* barometers, where a vendor has changed parts do to unavailability.

Reply:
> There are obviously limitations, but its a good starting point. I think ultimately we should force metadata to be put in hwdef that removes these issues, but that will have to evolve over time.

Well, I put in the "serial purpose" stuff in to see what that might look like.